(dp1
S'output'
p2
S'<class \'_mysql_exceptions.OperationalError\'> (1054, "Unknown column \'idPreg\' in \'where clause\'")'
p3
sS'layer'
p4
S'F:\\Informatorio\\web2py\\applications\\BC\\controllers/PreguntasController.py'
p5
sS'code'
p6
S'# -*- coding: utf-8 -*-\n# intente algo como\nfrom gluon.serializers import json\ndef index():\n   \tresponse.view=\'PreguntasController/index.html\'\n   \treturn dict()\n\t\ndef seleccionar_pregunta():\n\trespuesta={}\n\tresultado=[] #lista que almacenara el diccionario de categoria,pregunta y respuesta\n\timagen={}\n\tresul_categoria = db.executesql(\'SELECT id,descripcion FROM categoria ORDER BY RAND() LIMIT 1\')\n\tfor row in resul_categoria:\n \t\tcategoria = {\'idCat\': row[0], \'descCat\': row[1]}\n \n\tresul_pregunta = db.executesql(\'SELECT id,descripcion FROM pregunta WHERE idcat=\'+ str(categoria[\'idCat\']) +\' ORDER BY RAND() LIMIT 1\')\n\tfor row in resul_pregunta:\n\t \tpregunta = {\'idPreg\': row[0], \'descPreg\': row[1]}\n\n\tif categoria[\'idCat\']==4:\n\t\tbimg = db.executesql(\'SELECT id,imagen FROM escultura WHERE idPreg=\'+str(pregunta[\'idPreg\']))\n\t\tfor row in bimg:\n  \t\t\timagen={\'idImg\':row[0],\'imagen\':row[1]}\n \telse:\n \t\timagen={\'imagen\':""}\n\n\tresul_respuesta=db.executesql(\'SELECT id,descripcion FROM RESPUESTAPREGUNTA WHERE idPreg=\'+str(pregunta[\'idPreg\']))\n\tx=0\n\tfor row in resul_respuesta:\n\t \trespuesta[\'idResp\'+str(x)] = row[0] \n\t \trespuesta[\'descResp\'+str(x)] = row[1]\n\t \tx=x+1\n\n \tresultado.append(categoria)\n \tresultado.append(pregunta)\n \tresultado.append(respuesta)\n \tresultado.append(imagen)\n \treturn response.json(resultado)\n\n\ndef verificar_respuesta():\n\tidPreg= request.vars.idPreg\n\tresp_usuario= request.vars.resp\n\n\topcion=db.executesql(\'SELECT descripcion from respuestapregunta WHERE opcion="V" AND idPreg=\'+str(idPreg))\n\tfor resp in opcion:\n\t\tif str(resp[0])==resp_usuario:\n\t\t\trespuesta=\'V\'\n\t\telse:\n\t\t\trespuesta=\'F\'\n\n \treturn (respuesta)\n\ndef registrarPuntaje():\n\tpass\n\n\n\t\n\n\n\n\n\n\n\n\n\n\nresponse._vars=response._caller(seleccionar_pregunta)\n'
p7
sS'snapshot'
p8
(dp9
sS'traceback'
p10
S'Traceback (most recent call last):\n  File "F:\\Informatorio\\web2py\\gluon\\restricted.py", line 227, in restricted\n    exec ccode in environment\n  File "F:\\Informatorio\\web2py\\applications\\BC\\controllers/PreguntasController.py", line 69, in <module>\n  File "F:\\Informatorio\\web2py\\gluon\\globals.py", line 417, in <lambda>\n    self._caller = lambda f: f()\n  File "F:\\Informatorio\\web2py\\applications\\BC\\controllers/PreguntasController.py", line 21, in seleccionar_pregunta\n    bimg = db.executesql(\'SELECT id,imagen FROM escultura WHERE idPreg=\'+str(pregunta[\'idPreg\']))\n  File "F:\\Informatorio\\web2py\\gluon\\packages\\dal\\pydal\\base.py", line 1019, in executesql\n    adapter.execute(query)\n  File "F:\\Informatorio\\web2py\\gluon\\packages\\dal\\pydal\\adapters\\base.py", line 1388, in execute\n    return self.log_execute(*a, **b)\n  File "F:\\Informatorio\\web2py\\gluon\\packages\\dal\\pydal\\adapters\\base.py", line 1382, in log_execute\n    ret = self.get_cursor().execute(command, *a[1:], **b)\n  File "MySQLdb/cursors.py", line 205, in execute\n  File "MySQLdb/connections.py", line 36, in defaulterrorhandler\nOperationalError: (1054, "Unknown column \'idPreg\' in \'where clause\'")\n'
p11
s.